hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/execution/bulkinsert/RDDSimpleBucketBulkInsertPartitioner.java

```diff
@@ -93,7 +96,9 @@ Map<String, Map<Integer, String>> getPartitionMapper(JavaRDD<HoodieRecord<T>> re
           // Generate a file that does not exist
           for (int i = 0; i < numBuckets; i++) {
             if (!existsBucketID.contains(i)) {
-              String fileIdPrefix = BucketIdentifier.newBucketFileIdPrefix(i);
+              String fileIdPrefix = isNonBlockingConcurrencyControl
+                  ? BucketIdentifier.newBucketFileIdFixedSuffix(i)
+                  : BucketIdentifier.newBucketFileIdPrefix(i);
```

Comment:
Maybe we can have a unified method like

```java
BucketIdentifier.newBucketFileIdPrefix(String bucketId, boolean fixed)
```

So that we can inline the if ... else block inside the BucketIdentifier.

Reply:
Done